gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -480,14 +523,15 @@ EliminateHybrid(const HybridGaussianFactorGraph &factors,
         std::inserter(continuousSeparator, continuousSeparator.begin()));
 
     // Similarly for the discrete separator.
-    KeySet discreteSeparatorSet;
-    std::set<DiscreteKey> discreteSeparator;
     auto discreteKeySet = factors.discreteKeySet();
+    // Use set-difference to get the difference in keys
```

Comment:
Improve comment. Don’t understand what is happening without reading the code. Why are we doing this? Explain in comment.

Reply:
I refactored this since we don't need a set difference. All the discrete keys should be the discrete separator when doing `hybridElimination`.